src/trackedit/internal/au3/au3interaction.cpp

```diff
@@ -891,6 +893,145 @@ void Au3Interaction::duplicateTracks(const TrackIdList& trackIds)
     projectHistory()->pushHistoryState("Duplicate track", "Duplicate track");
 }
 
+void Au3Interaction::moveTracksUp(const TrackIdList& trackIds)
+{
+    for (const auto trackId : trackIds) {
+        moveTrack(trackId, TrackMoveDir::Up);
+    }
+    projectHistory()->pushHistoryState("Move track", "Move track");
+}
+
+void Au3Interaction::moveTracksDown(const TrackIdList& trackIds)
+{
+    for (const auto trackId : trackIds) {
+        moveTrack(trackId, TrackMoveDir::Down);
+    }
+    projectHistory()->pushHistoryState("Move track", "Move track");
+}
+
+void Au3Interaction::moveTracksToTop(const TrackIdList& trackIds)
+{
+    for (const auto trackId : trackIds) {
+        moveTrack(trackId, TrackMoveDir::Top);
+    }
+    projectHistory()->pushHistoryState("Move track", "Move track");
+}
+
+void Au3Interaction::moveTracksToBottom(const TrackIdList& trackIds)
+{
+    for (const auto trackId : trackIds) {
+        moveTrack(trackId, TrackMoveDir::Bottom);
+    }
+    projectHistory()->pushHistoryState("Move track", "Move track");
+}
+
+void Au3Interaction::moveTracksTo(const TrackIdList& trackIds, int to)
+{
+    for (const auto trackId : trackIds) {
+        moveTrackTo(trackId, to++);
+    }
+    projectHistory()->pushHistoryState("Move track", "Move track");
+}
+
+void Au3Interaction::moveTrackTo(const TrackId trackId, int to)
+{
+    auto& project = projectRef();
+    auto& tracks = ::TrackList::Get(project);
+    ::Track* au3Track = DomAccessor::findTrack(project, ::TrackId(trackId));
+
+    IF_ASSERT_FAILED(au3Track) {
+        return;
+    }
+
+    int from = std::distance(tracks.begin(), tracks.Find(au3Track));
+
+    if (to == from) {
+        return;
+    }
+
+    int pos = from;
+    if (pos < to) {
+        while (pos != to && tracks.CanMoveDown(*au3Track)) {
+            pos++;
+            tracks.MoveDown(*au3Track);
+        }
+    } else {
+        while (pos != to && tracks.CanMoveUp(*au3Track)) {
+            pos--;
+            tracks.MoveUp(*au3Track);
+        }
+    }
+
+    if (pos != to) {
+        LOGW() << "Can't move track to " << to;
```

Comment:
It would be useful to know the current position of the track (pos) too

Reply:
absolutely